Lib/typemaps/strings.swg

```diff
@@ -427,70 +426,27 @@
 %typemap(freearg,noblock=1,match="in") (Char *STRING, size_t LENGTH) {
   if (alloc$argnum == SWIG_NEWOBJ) SWIG_DeleteCharArray(buf$argnum);
 }
-/* old 'int' form */
-%typemap(in) (Char *STRING, int LENGTH) = (Char *STRING, size_t LENGTH);
-%typemap(freearg) (Char *STRING, int LENGTH) = (Char *STRING, size_t LENGTH);
-
-
-/* Here size includes the '0' terminator */
-%typemap(in,noblock=1,fragment=#SWIG_AsCharPtrAndSize)
-  (Char *STRING, size_t SIZE) (int res, Char *buf = 0, size_t size = 0, int alloc = 0),
-  (const Char *STRING, size_t SIZE) (int res, Char *buf = 0, size_t size = 0, int alloc = 0)
+%typemap(directorin,noblock=1,fragment=#SWIG_FromCharPtrAndSize)
+    (Char *STRING, size_t LENGTH)
 {
-  res = SWIG_AsCharPtrAndSize($input, &buf, &size, &alloc);
-  if (!SWIG_IsOK(res)) {
-    %argument_fail(res,"$type",$symname, $argnum);
+  if ($1 && $2 > 0) {
+    $input = SWIG_FromCharPtrAndSize((const char*)$1, (size_t)$2);
+  } else {
+    $input = SWIG_FromCharPtrAndSize("", 0);
   }
-  $1 = %reinterpret_cast(buf, $1_ltype);
-  $2 = %numeric_cast(size, $2_ltype);
 }
-%typemap(freearg,noblock=1,match="in") (Char *STRING, size_t SIZE) {
-  if (alloc$argnum == SWIG_NEWOBJ) SWIG_DeleteCharArray(buf$argnum);
-}
-/* old 'int' form */
-%typemap(in) (Char *STRING, int SIZE) = (Char *STRING, size_t SIZE);
-%typemap(freearg) (Char *STRING, int SIZE) = (Char *STRING, size_t SIZE);
 
+%typemap(in) (const Char *STRING, size_t LENGTH) = (Char *STRING, size_t LENGTH);
+%typemap(freearg) (const Char *STRING, size_t LENGTH) = (Char *STRING, size_t LENGTH);
+%typemap(directorin) (const Char *STRING, size_t LENGTH) = (Char *STRING, size_t LENGTH);
 
-/* reverse order versions */
```

Comment:
The reverse order versions seem to have been lost. Was that intentional? They'd need putting back in.

Reply:
OK.
It was not tested, not documented and not implemented in any of the other  languages.
I will put it back and add it to all languages.